src/main/java/frc/utils/Vector.java

```diff
@@ -0,0 +1,422 @@
+package frc.utils;
+
+import java.util.function.Consumer;
+
+// a class representing either a vector or a Point in space in 2 or more dimensions with double
+// precision
+// with operations for manipulating them
+
+// most operations affect the point they're called on, so for non-destructive operations, use
+// .copy()
+
+// when clockwise or counterclockwise is mentioned, keep in mind that in java, positive y is down,
+// not up
+public class Vector {
```

Comment:
A bit misleading to use. Can you separate this class into a point class and a vector class?